src/trait_/common_ord.rs

```diff
@@ -49,114 +57,153 @@ pub fn build_ord_signature(item: &Item, trait_: &DeriveTrait, body: &TokenStream
 				}
 			};
 
-			// Nightly or unsafe (default) implementation.
-			#[cfg(any(feature = "nightly", not(feature = "safe")))]
-			{
-				let path = trait_.path();
-				let method = match trait_ {
-					PartialOrd => quote! { partial_cmp },
-					Ord => quote! { cmp },
-					_ => unreachable!("unsupported trait in `prepare_ord`"),
-				};
+			let incomparable = build_incomparable_pattern(variants);
 
-				// Nightly implementation.
-				#[cfg(feature = "nightly")]
+			// If there is only one comparable variant, it has to be it when it is non
+			// incomparable.
+			let mut comparable = variants.iter().filter(|v| !v.is_incomparable());
+			if let (Some(comparable), None) = (comparable.next(), comparable.next()) {
```

Comment:
This looks very weird :confused:.
I know it's not as clean as you did it, but I find this clearer:
```suggestion
			if comparable.len() == 1 {
				let comparable = comparable.next().expect("there should be one variant");
```
Feel free to ignore my suggestion if you don't like it.

Reply:
Well it does improve readability, so I think it does make sense. Sometimes I try to be too clever :D

Reply:
Actually, not that simple, comparable is an iterator if we don't want to arbitrarily collect it prior, the easiest thing is to just add a comment explaining instead.

Reply:
`.len()` doesn't collect it prior? I'm not following.

Reply:
you mean `.count()`? But we need both the first element and ensure that there is no second.

Your proposed code would be this:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a06ad92d355d1908ea244c677bbe60ca

Reply:
Ah, [`Filter`](https://doc.rust-lang.org/std/iter/struct.Filter.html) doesn't implement [`ExactSizeIterator`], obviously, that's why it doesn't have [`.len()`](https://doc.rust-lang.org/std/iter/trait.ExactSizeIterator.html#method.len).

Thanks for the explanation.